src/cp/cp_api.c

```diff
@@ -250,18 +251,22 @@ __wt_conn_control_point_enable(WT_CONNECTION *wt_conn, wt_control_point_id_t id,
  */
 int
 __wti_session_control_point_enable(
-  WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry, const char *cfg)
+  WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry, const char *extra_cfg)
 {
+    WT_CONNECTION_IMPL *conn;
     WT_CONTROL_POINT_DATA *cp_data;
-    const char *cfgs[2];
+    const char *cfgs[3];
 
+    conn = S2C(session);
     cp_data = cp_registry->cp_data;
     if (WT_UNLIKELY(cp_data != NULL))
         /* Already enabled. */
         return (EEXIST);
-    cfgs[0] = cfg;
-    cfgs[1] = NULL;
-    cp_data = cp_registry->init(session, cp_registry->config_name, cfgs);
+    cfgs[0] = conn->cfg;
```

Comment:
Since you have renamed the function parameter to extra_cfg, rename cfgs[] to cfg][.